src/include/tcop/tcop.h

```diff
@@ -73,8 +73,42 @@ class TrafficCop {
   // InitBindPrepStmt - Prepare and bind a query from a query string
   std::shared_ptr<Statement> PrepareStatement(const std::string &statement_name,
                                               const std::string &query_string,
+                                              std::string &error_message,
+                                              const size_t thread_id = 0);
+
+  // PortalExec - Execute query string
+  ResultType ExecuteStatementJDBC(const std::string &query,
```

Comment:
probably rename it as `ExecuteStatementExtended` or something else? The `ExecuteStatement` can also be invoked by jdbc.